fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownFilterThroughPTopN.java

```diff
@@ -0,0 +1,93 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
+import org.apache.doris.nereids.trees.plans.logical.LogicalPartitionTopN;
+
+import com.google.common.collect.Sets;
+
+import java.util.Set;
+
+/**
+ * Push down the 'filter' pass through the 'partitionTopN' if filter key is partitionTopN's partition key.
+ * Logical plan tree:
+ *                 any_node
+ *                   |
+ *                filter (a <= 100)
+ *                   |
+ *                partition topn (PARTITION BY a)
+ *                   |
+ *                 any_node
+ * transformed to:
+ *                 any_node
+ *                   |
+ *                partition topn (PARTITION BY a)
+ *                   |
+ *                filter (a <= 100)
+ *                   |
+ *                 any_node
+ */
+
+public class PushdownFilterThroughPTopN extends OneRewriteRuleFactory {
+
+    @Override
+    public Rule build() {
+        return logicalFilter(logicalPartitionTopN()).thenApply(ctx -> {
+            LogicalFilter<LogicalPartitionTopN<Plan>> filter = ctx.root;
+            LogicalPartitionTopN<Plan> partitionTopN = filter.child();
+            // follow the similar checking and transformation rule as
+            // PushdownFilterThroughWindow
+            Set<Expression> bottomConjuncts = Sets.newHashSet();
+            Set<Expression> upperConjuncts = Sets.newHashSet();
+            for (Expression expr : filter.getConjuncts()) {
+                boolean pushed = false;
+                for (Expression partitionKey : partitionTopN.getPartitionKeys()) {
+                    if (partitionKey instanceof SlotReference
+                            && partitionKey.getInputSlots().containsAll(expr.getInputSlots())) {
```

Comment:
"partitionKey.getInputSlots().containsAll(expr.getInputSlots())"
is this condition ok? 
in following case, a<10 cannot be pushed down
filter(a<10) 
+----partition by (a + b)

Reply:
partition key should be slot，so this plan chould not appear. if we write a + b as partition key in sql. we will get plan like this:
```
project(except(c))
+-- filter(a < 10)
    +-- window(... partition by (c))
        +-- project(a, a + b as c, ...) 
```

Reply:
it's ok since for a+b the input slot size is 1 and the slot is a+b, not a and b. So it can prevent this kind of unexpected push down. And another thing is the rule is followed by pushFilterThroughWindow, it should be valid since this kind of checking has been done at  pushFilterThroughWindow. The similar process here is just maintain the dependence between rules but commonly these two rules should appear at the same time.